### ISSUE TITLE
Build errors with latest autoconf and autoheader

### ISSUE THREAD

**Issue by hanakp-VUT** [dgtlrift/hp2xx#8]
Hello, I know it's probably related to other open issues, but I'll mention it anyway. I tried to build the project on latest Ubuntu 24.04.1 LTS virtual machine. It uses newer autoconf and autoheader versions than hp2xx, so I used autoreconf and autoupdate as suggested. However, it didn't help and make failed with 2 errors and many warnings. It's been a loooong time I did anything in C, but it seems there are problems with the source code itself, not with autoconf or autoheader. I tried it two times with slightly different command order, but the build errors remained the same. I'm attaching entire console dump from the first attempt.

U also tried fork from [https://github.com/fenugrec/hp2xx](https://github.com/fenugrec/hp2xx) hoping they fixed the problems, but the result was the same.

[console dump.txt](https://github.com/user-attachments/files/17375841/console.dump.txt)

**Comment by andrewbird** [comment]
Hi there, I started to fix this but then noticed that somebody else already has started working on it over at https://github.com/fenugrec/hp2xx/tree/buildsys_fixes  (I just noticed that you @hanakp-VUT, had also noticed and tried it). @fenugrec are you still working on it, will you be submitting a PR?

**Comment by hanakp-VUT** [comment]
Actually, I haven't noticed the buildsys_fixes branch in @fenugrec repository. But I tried compiling it now and it produces the same errors as master branch.

**Comment by andrewbird** [comment]
I added PR #9 with an autoconf update and the bare minimum fixes from fenugrec. This builds ~~fine~~ with warnings on my Ubuntu 24.04.1 LTS.

**Comment by hanakp-VUT** [comment]
Many thanks, it works now!

**Comment by fenugrec** [comment]
I... don't remember where I left off, busy for a few days, but I haven't totally abandoned. I was mostly waiting for signs of life from project maintainer(s)

**Comment by jquertin** [comment]
I'm also running into the issue. Thanks for the fix!

**Comment by fenugrec** [comment]
@andrewbird , @hanakp-VUT , are you all still using this and interested ? maybe we can make a shared repo to make it easier to keep track of each other's build fixes... I might even add some features ( I want my images with a black background...)

**Comment by andrewbird** [comment]
Hi @fenugrec, I'm not actually using it myself any more, but don't mind contributing as and when I see something I can fix.

**Comment by hanakp-VUT** [comment]
I don't use it anymore, either. Neverheless, it would be nice if could include andrewbird's fixes into the master, so others don't run into the same problem. There are not many HPGL utilities out there. Thanks!

**Comment by fenugrec** [comment]
fair enough ! I have some instruments that only speak hpgl so I'll be using hp2xx for a while yet. 

andrew, do you remember what fixes you needed, that weren't in my build fixes branch ?
Compiling here works fine (gcc 15.2.1 on arch)

I'm also going to ping the original maintainer (Martin) again, although I haven't heard from him in a while.

**Comment by andrewbird** [comment]
The PR is still here #9 , but essentially three picked from yourself and an updated autoconf. If you have it working already on Arch, then that's as good a start as any.

**Comment by fenugrec** [comment]
> essentially three picked from yourself and an updated autoconf.

ah yes, right. I was never a fan of autotools and this autoreconf autoupdate stuff... constant aggravation

**Comment by hanakp-VUT** [comment]
@fenugrec since you're here, I'd like to ask about one thing. I have a pen plotter that doesn't know polygon commands, so filled areas have to be drawn as many parallel lines in HPGL. Back in 2024, I tried hp2xx to convert polygon HPGLs into line HPGLs. But hp2xx  didn't obey pen thickness setting and the output files had hundreds of lines with very fine spacing. That was unusable for plotting. Is that expected behavior? Or is there some way around it? I described it and put a small test HPGL here:

https://www.eevblog.com/forum/general-computing/hpgl-language-sw-to-convert-polygons-to-set-of-parallel-lines/msg5679119/#msg5679119

**Comment by fenugrec** [comment]
>But hp2xx didn't obey pen thickness setting and the output files had hundreds of lines with very fine spacing.

Heh that is quite beyond my knowledge of hpgl and hp2xx, sorry ! Have you tried the 'latest' a37 package from http://ruby.chemie.uni-freiburg.de/~martin/hp2xx-3.5.0a37.tar.gz ? that tarballl was never merged into this repo, and I understand it has some Fill tweaks; you may be lucky...

**Comment by fenugrec** [comment]
I haven't had a reply, yet Martin seems to be quite active... can someone else try contacting him in case my emails are just bouncing ? I tried his uni-freiburg email address.

**Comment by andrewbird** [comment]
You do know he has a github account 'martin-frbg'?

**Comment by fenugrec** [comment]
> You do know he has a github account 'martin-frbg'?

Yes, that's what I mean by 'he seems to be active'

**Comment by andrewbird** [comment]
Ahh, okay!
